include/oneapi/dpl/pstl/glue_algorithm_ranges_impl.h

```diff
@@ -318,6 +318,249 @@ struct __search_n_fn
+        if constexpr(!std::ranges::sized_range<_R1> || !std::ranges::sized_range<_R2>)
+            return false;
+        else
+        {
+            const auto __dispatch_tag = oneapi::dpl::__ranges::__select_backend(__exec);
+            return oneapi::dpl::__internal::__ranges::__pattern_equal(__dispatch_tag,
+                std::forward<_ExecutionPolicy>(__exec), std::forward<_R1>(__r1), std::forward<_R2>(__r2), __pred,
+                __proj1, __proj2);
+        }
+    }
+}; //__equal_fn
+}  //__internal
+
+inline constexpr __internal::__equal_fn equal;
+
+namespace __internal
+{
+struct __is_sorted_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _R, typename _Proj = std::identity,
+             std::indirect_strict_weak_order<std::projected<std::ranges::iterator_t<_R>, _Proj>>
+             _Comp = std::ranges::less>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>> && std::ranges::sized_range<_R>
+    bool
+    operator()(_ExecutionPolicy&& __exec, _R&& __r, _Comp __comp = {}, _Proj __proj = {}) const
+    {
+        const auto __dispatch_tag = oneapi::dpl::__ranges::__select_backend(__exec);
+        return oneapi::dpl::__internal::__ranges::__pattern_is_sorted(__dispatch_tag,
+            std::forward<_ExecutionPolicy>(__exec), std::forward<_R>(__r), __comp, __proj);
+    }
+}; //__is_sorted_fn
+}  //__internal
+
+inline constexpr __internal::__is_sorted_fn is_sorted;
+
+namespace __internal
+{
+struct __stable_sort_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _R, typename _Comp = std::ranges::less,
+             typename _Proj = std::identity>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>> && std::ranges::sized_range<_R>
+           && std::sortable<std::ranges::iterator_t<_R>, _Comp, _Proj>
+    auto
+    operator()(_ExecutionPolicy&& __exec, _R&& __r, _Comp __comp = {}, _Proj __proj = {}) const
+    {
+        const auto __dispatch_tag = oneapi::dpl::__ranges::__select_backend(__exec);
+        return oneapi::dpl::__internal::__ranges::__pattern_sort_ranges(__dispatch_tag,
+            std::forward<_ExecutionPolicy>(__exec), std::forward<_R>(__r), __comp, __proj);
+    }
+}; //__stable_sort_fn
+}  //__internal
+
+inline constexpr __internal::__stable_sort_fn stable_sort;
+
+namespace __internal
+{
+struct __sort_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _R, typename _Comp = std::ranges::less,
+             typename _Proj = std::identity>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>> && std::ranges::sized_range<_R>
+           && std::sortable<std::ranges::iterator_t<_R>, _Comp, _Proj>
+    auto
+    operator()(_ExecutionPolicy&& __exec, _R&& __r, _Comp __comp = {}, _Proj __proj = {}) const
+    {
+        return oneapi::dpl::ranges::stable_sort(std::forward<_ExecutionPolicy>(__exec), std::forward<_R>(__r),
```

Comment:
I am not sure if we should call `stable_sort` for `sort`, at least on that layer. It may be justified on device backend layer since there is only stable implementation, but host patterns may benefit from calling `sort` from the performance perspective.

Reply:
I've added a TODO comment in order not to forget about it.